x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/SecurityImplicitBehaviorBootstrapCheckTests.java

```diff
@@ -32,8 +36,13 @@ public void testFailureUpgradeFrom7xWithImplicitSecuritySettings() throws Except
         );
         NodeMetadata nodeMetadata = new NodeMetadata(randomAlphaOfLength(10), previousVersion, Version.CURRENT);
         nodeMetadata = nodeMetadata.upgradeToCurrentVersion();
-        BootstrapCheck.BootstrapCheckResult result = new SecurityImplicitBehaviorBootstrapCheck(nodeMetadata).check(
-            createTestContext(Settings.EMPTY, createLicensesMetadata(previousVersion, randomFrom("basic", "trial")))
+        ClusterStateLicenseService licenseService = mock(ClusterStateLicenseService.class);
+        Metadata metadata = createLicensesMetadata(previousVersion, randomFrom("basic", "trial"));
+        License license = mock(License.class);
+        when(licenseService.getLicense(metadata)).thenReturn(license);
+        when(license.operationMode()).thenReturn(randomFrom(License.OperationMode.BASIC, License.OperationMode.TRIAL));
```

Comment:
I think it's simpler and provide more test coverage with
```suggestion
        when(licenseService.getLicense(metadata)).thenCallRealMethod();
```

Reply:
That won't work since since we also need to call `license.operationMode()`. The return value from `.thenCallRealMethod()` is an `OngoingStubbing<License>`. While the real method is indeed called, the real object (`License`) is not returned and the test will fail with `Cannot invoke "org.elasticsearch.license.License.operationMode()" because "license" is null`. I couldn't find a quick solution online and in general I try to avoid using partial mocks (i.e. callRealMethod and spy) since they generally test more that just the unit you are trying to test.

Reply:
I see this is because `getLicense` internally calls `getLicenseFromLicensesMetadata` which isn't mocked. It would work if `getLicenseFromLicensesMetadata` is inlined. I am not suggesting that, mainly trying to understand why it didn't work. Feel free to keep how it is.